### PR TITLE
AEA-1822: Show errors in the EPSAT client.

### PR DESCRIPTION
**Pull request by mfjarvis**

### CODE REVIEW

File: tool/site/client-react/src/components/axiosResponseView.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import {AxiosResponse} from "axios"
+import {SummaryList} from "nhsuk-react-components"
+import React from "react"
+import {MessageExpander} from "./messageExpanders"
+
+interface AxiosResponseViewProps {
+  response: AxiosResponse
+}
+
```
[the remark this answers is not in the document]
Comment (booshibooshi) [reply tool/site/client-react/src/components/axiosResponseView.tsx:9]: I have tested with a 429 error and it displays the error nicely, wondering if we could also include request headers (in particular `x-request-id` for debugging), request body and url for the backend. What do you think?

image

File: tool/site/client-react/src/requests/result.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import {FhirResource} from "fhir/r4"
+
+export function isResult(data: unknown): data is Result {
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply tool/site/client-react/src/requests/result.ts:3]: Yeah I didn't like the name anyway :+1:

File: tool/site/client-react/src/components/axiosResponseView.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mfjarvis) [reply tool/site/client-react/src/components/axiosResponseView.tsx]: Added request details and decoupled from Axios

[the remark this answers is not in the document]
Comment (mfjarvis) [reply tool/site/client-react/src/components/axiosResponseView.tsx]: Request details don't seem to work. Not sure axios lets you get at them from the response

[the remark this answers is not in the document]
Comment (mfjarvis) [reply tool/site/client-react/src/components/axiosResponseView.tsx]: User error, got it working

File: tool/site/client-react/src/pages/sendPreSignPage.tsx [hunk 3]
```diff
@@ -81,13 +83,14 @@ const SendPreSignPage: React.FC<SendPreSignPageProps> = ({
...
+  const response = await axiosInstance.get<Bundle | OperationOutcome>(`${baseUrl}prescription/${prescriptionId}`)
+  return getResponseDataIfValid(response, isBundle)
 }
 
 async function sendSignRequest(baseUrl: string) {
-  const response = await axios.post<SignResponse>(`${baseUrl}prescribe/sign`)
-  const prepareErrors = response.data.prepareErrors
+  const response = await axiosInstance.post<SignResponse>(`${baseUrl}prescribe/sign`)
+  const signResponse = getResponseDataIfValid(response, isSignResponse)
+  const prepareErrors = signResponse.prepareErrors
```
[the remark this answers is not in the document]
Comment (mfjarvis) [reply tool/site/client-react/src/pages/sendPreSignPage.tsx:94]: New PR?

**Review by dan-jones-iw — APPROVED** [review]

**Pull request merged**